Sources/SPMBuildCore/PluginScriptRunner.swift

```diff
@@ -0,0 +1,115 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021-2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Basics
+import Foundation
+import PackageModel
+import PackageLoading
+import PackageGraph
+import TSCBasic
+import struct TSCUtility.Triple
+
+/// Implements the mechanics of running and communicating with a plugin (implemented as a set of Swift source files). In most environments this is done by compiling the code to an executable, invoking it as a sandboxed subprocess, and communicating with it using pipes. Specific implementations are free to implement things differently, however.
+public protocol PluginScriptRunner {
```

Comment:
This factored out from PluginInvocation.swift and deals just with the contract between the plugin invocation and the entity that can compile, run, and talk to the plugin on any given platform (of which DefaultPluginScriptRunner is the stock implementation).